projects/storefrontapp-e2e-cypress/cypress/helpers/cart.ts

```diff
@@ -131,16 +164,17 @@ export function removeAllItemsFromCart() {
 
   cy.wait('@refresh_cart');
 
-  cy.get('cx-cart-details .cx-total').should('contain', 'Cart #');
-
   getCartItem(product1.name).within(() => {
     cy.getByText('Remove').click();
   });
 
-  cy.get('.EmptyCartMiddleContent').should(
-    'contain',
-    'Your shopping cart is empty'
-  );
+  validateEmptyCart();
+}
+
+export function removeCartItem(product) {
```

Comment:
You noticed some really nice code/idea.

Write something nice to author.

eg
> That's very elegant solution to this problem. Well done.